### PR TITLE
Add margin variable for paragraphs

### LINKED ISSUE #23031: [v4] Introduce variable for customizing margin-bottom for "p"
_reboot.scss has:

```
p {
  margin-top: 0;
  margin-bottom: 1rem;
}
```

However the value is hardcoded. It would be nice to have it exported to _variables.scss so that people can customize (_custom.scss was nice but looks like it's removed now).

### PR DESCRIPTION
**Pull request by prateekgoel**
I added a section for paragraphs in _variables.scss, used that variable to change hard coded margin value for p element.

Fixes #23031

### CODE REVIEW

**Review by mdo — CHANGES_REQUESTED** [review]
Paragraphs don't have horizontal margin on them (by default or from Bootstrap), so there's no need to reset them if there is nothing there to begin with.

Please restore the `margin-top: 0;` and use a variable for the `margin-bottom`.

Also, no need to shorten the word to four letters—use `$paragraph-margin-bottom` or something like that maybe?

**Comment by prateekgoel** [comment]
Ok. I'll do these changes. Thanks.

**Comment by prateekgoel** [comment]
@mdo For now, I have restored `margin-top: 0;`
Let me know if anything else need to be done.

**Comment by prateekgoel** [comment]
@mdo Any update on this?

File: scss/_variables.scss [hunk 1]
```diff
@@ -787,6 +794,8 @@ $breadcrumb-padding-y:          .75rem !default;
 $breadcrumb-padding-x:          1rem !default;
 $breadcrumb-item-padding:       .5rem !default;
 
+$breadcrumb-margin:             0 0 1rem !default;
```
Comment (mdo) [inline scss/_variables.scss]: This should be here

**Review by mdo — CHANGES_REQUESTED** [review]
This'll go in for Beta 2 at the earliest.

File: scss/_variables.scss (hunk 1 above)
Comment (prateekgoel) [inline scss/_variables.scss]: @mdo Should I add breadcumb-margin too? I mean this PR is for paragraph-margin. For breadcrumb there's a different PR #23138 .

Comment (mdo) [inline scss/_variables.scss]: Crap, I typo-ed—I meant shouldn't be here. It's in the wrong PR, so you can remove it.

Comment (prateekgoel) [inline scss/_variables.scss]: @mdo Did the changes.

**Pull request merged**